src/metabase/xrays/transforms/core.clj

```diff
@@ -69,26 +69,34 @@
 (defn- maybe-add-expressions
   [bindings {:keys [expressions name]} query]
   (if expressions
-    (-> query
-        (assoc :expressions (->> expressions
-                                 keys
-                                 (select-keys (get-in bindings [name :dimensions]))))
-        (update :fields concat (for [expression (keys expressions)]
-                                 [:expression expression])))
+    (let [expr-clauses (->> expressions
+                            keys
+                            (select-keys (get-in bindings [name :dimensions])))]
+      (-> query
+          (assoc :expressions       expr-clauses
+                 :expression-idents (update-vals expr-clauses (fn [_] (u/generate-nano-id))))
+          (update :fields concat (for [expression (keys expressions)]
+                                   [:expression expression]))))
     query))
 
+(defn- indexed-idents [seqable]
+  (when (seq seqable)
+    (into {} (map (fn [i] [i (u/generate-nano-id)]))
+          (range (count seqable)))))
```

Comment:
This is at least the third way of writing this. :)

Reply:
I agree... it feels like this should be a built-in, sort of like `frequencies`.

Reply:
I've consolidated these with a helper.